packages/contracts/test/verifier.test.ts

```diff
@@ -118,6 +119,74 @@ describe('Epoch key proof verifier', function () {
         }
     })
 
+    it('should decode public signals', async () => {
+        const accounts = await ethers.getSigners()
+        const attester = accounts[1]
+        const id = new ZkIdentity()
+        // sign up a user
+        const { leaf, index, epoch } = await signupUser(
+            id,
+            unirepContract,
+            attester.address,
+            attester
+        )
+        stateTree.insert(leaf)
+
+        const merkleProof = stateTree.createProof(index)
+        const posRep = 0
+        const negRep = 0
+        const graffiti = 0
+        const timestamp = 0
+        const data = 0
+        for (let nonce = 0; nonce < NUM_EPOCH_KEY_NONCE_PER_EPOCH; nonce++) {
+            const r = await defaultProver.genProofAndPublicSignals(
+                Circuit.verifyEpochKey,
+                stringifyBigInts({
+                    state_tree_elements: merkleProof.siblings,
+                    state_tree_indexes: merkleProof.pathIndices,
+                    identity_nullifier: id.identityNullifier,
+                    pos_rep: posRep,
+                    neg_rep: negRep,
+                    graffiti,
+                    timestamp,
+                    data,
+                    control: EpochKeyProof.buildControlInput({
+                        epoch: epoch.toNumber(),
+                        nonce,
+                        attesterId: attester.address,
+                        revealNonce: 0,
+                    }),
+                })
+            )
+
+            const v = await defaultProver.verifyProof(
+                Circuit.verifyEpochKey,
+                r.publicSignals,
+                r.proof
+            )
+            expect(v).to.be.true
+            const proof = new EpochKeyProof(r.publicSignals, r.proof)
+            const signals = await unirepContract.decodeEpochKeySignals(
```

Comment:
should also test `decodeReputationSignals`

Reply:
[a585690](https://github.com/Unirep/Unirep/pull/216/commits/a5856906cb6aea95f3349e93adbafcaa32f415c1)